deployment/postgres-init-db/sql/tables/scheduler/scheduling_specification/scheduling_specification_goals.sql

```diff
@@ -1,14 +1,16 @@
 create table scheduler.scheduling_specification_goals (
+  goal_invocation_id integer generated by default as identity,
   specification_id integer not null,
   goal_id integer not null,
   goal_revision integer, -- latest is null
   priority integer not null,
   enabled boolean not null default true,
+  arguments jsonb not null default '{}'::jsonb,
 
   simulate_after boolean not null default true,
 
   constraint scheduling_specification_goals_primary_key
-    primary key (specification_id, goal_id),
+    primary key (specification_id, goal_id, goal_invocation_id),
```

Comment:
`goal_invocation_id` is sufficient to be the primary key, as it's a generated column. If that's not intended (as in, each goal on a spec should have its invocation id start from 0), then what `goal_definition.revision` does would be a better alternative.

Reply:
A side-effect of `goal_invocation_id` being the pkey is that the analysis tables can have their pkeys simplified as well

Reply:
Yeah I definitely thought about changing all the PKs in these tables, but figured the overspecified composite key was the least invasive way to introduce this invocation idea

Reply:
I think I agree with Theresa. Generally using the minimal primary key is best practice if you can

Reply:
Change has been made.

### Primary Key changes

`scheduling_specification_goals`:
```sql
primary key (goal_invocation_id),
```

`goal_analysis`:
```sql
primary key (analysis_id, goal_invocation_id)
```

`goal_analysis_created_activities`:
```sql
primary key (analysis_id, goal_invocation_id, activity_id)
```

`goal_analysis_satisfying_activities`:
```sql
primary key (analysis_id, goal_invocation_id, activity_id)
```

### Foreign Key changes 
`created_activites` and `satisfying_activities` now FK to `goal_analysis` instead of just `scheduling_run`, since `goal_analysis` holds the "as run" data for that goal invocation (e.g. arguments, revision).

We were duplicating this data into each of `created_activites` and `satisfying_activites` before (i.e. storing `goal_revision`, `goal_id`, etc), but I think with the addition of arguments, it makes sense to centralize this into a `goal_analysis` entry per goal invocation, and then have `created_activites` and `satisying_activities` be pure association tables between directives and a `goal_analysis` entry.

I believe this is how we had hasura metadata set up anyways, this is just codifying this into DB FKs and simplifying this `scheduling_run` sub-schema

![Untitled-2023-11-30-1236](https://github.com/user-attachments/assets/abbe5576-a4dd-4bc7-8c28-6947fe4e68ec)

Reply:
I've flagged the PR as "breaking" due to these PK changes